### ISSUE TITLE
train.py pascal voc dataset, 3 classes

### ISSUE THREAD

**Issue by hod1** [qqwweee/keras-yolo3#29]
Hi, i wrote an mjpeg stream -> object detection -> mjpeg stream service using your yolo object class, it works perfectly fine with 12-15 fps. I would like to limit the detection to 3 classes (person, bicycle, car), i tried to train a new model with these 3 classes, using your voc_annotation.py and train.py. Modified the voc_annotation.py:
classes = ["person","bicycle","car"]
Then removed rows without annotations from 2007_test.txt, 2007_train.txt and 2007_val.txt.
The original could detect cars from far away, but the new model could not detect them, only when they were already sideways. What should i do to increase the detection range without lowering score? (its already at 0.3)

**Comment by qqwweee** [comment]
When training data is not much, predicted box confidence is somewhat low, then leads to a low score. The reason is not figured out yet.
If you just want to clip the classes, clip the weights and load it on new model.

**Comment by hod1** [comment]
Hi, thanks for the advice, but i dont know how to do it.
#2. If i want to retrain these 3 new classes into the model (nothing else) should i use my modified CFG when using the "**python convert.py yolov3.cfg yolov3.weights model_data/yolo.h5**" command? Because the old cfg has 80 classes and 255 filters

**Comment by qqwweee** [comment]
If you have a model that can detect more than 3 classes ["person","bicycle","car"], then you can clip the weights to just detect these 3 classes. That is, output conv layers have 3\*(5+num_classes) filters, just use `model.get_weights()` to get the numpy array, choose some dims and load it on new model.
It is OK because yolo3 uses sigmoid as class probability function instead of softmax on yolo2.

By the way, for your 3 class training, the result is not satisfactory maybe because it needs a longer training.

**Comment by Amerry-c** [comment]
@hod1 I meet the same problem,and i dont understand how to load it on new model, did you solve this problem?

**Comment by qqwweee** [comment]
if  `my_classes` in `coco_classes`,
```
yolo = load_model('model_data/yolo.h5')
my_yolo = yolo_body(Input(shape=(None, None, 3)), 3, len(my_classes))

index = [0,1,2,3,4] + [coco_classes.index(i)+5 for i in my_classes]
w = [[item[..., index+[i+85 for i in index]+[i+170 for i in index]]
      for item in yolo.layers[i].get_weights()] for i in range(-3,0)]

for i in range(len(y.layers)-3):
    my_yolo.layers[i].set_weights(yolo.layers[i].get_weights())
for i,j in enumerate(range(-3,0)):
    my_yolo.layers[j].set_weights(w[i])
```

**Comment by someonelikeypp** [comment]
@qqwweee 
Hi, I have some problem need your help.

1. First is the same problem about this, if I what should I clip the weights, whether just modified the classes in yolov3.cfg:
![image](https://user-images.githubusercontent.com/15836287/41150796-7251344e-6b41-11e8-8a32-6ed954496a44.png)
Is there anything else I need to do? should I change output conv layers in yolov3.cfg, if there are 3 classes, output conv layers have 3*(5+3) = 24 filters? 

2. Second problem is: 
What is the use of these two files? (2007_val.txt, 2007_test.txt), I just use the 2007_train.txt.

thanks.
